modules/core/include/opencv2/core.hpp

```diff
@@ -819,12 +819,36 @@ mixChannels , or split .
 @param minLoc pointer to the returned minimum location (in 2D case); NULL is used if not required.
 @param maxLoc pointer to the returned maximum location (in 2D case); NULL is used if not required.
 @param mask optional mask used to select a sub-array.
-@sa max, min, compare, inRange, extractImageCOI, mixChannels, split, Mat::reshape
+@sa max, min, reduceMinMax, compare, inRange, extractImageCOI, mixChannels, split, Mat::reshape
 */
 CV_EXPORTS_W void minMaxLoc(InputArray src, CV_OUT double* minVal,
                             CV_OUT double* maxVal = 0, CV_OUT Point* minLoc = 0,
                             CV_OUT Point* maxLoc = 0, InputArray mask = noArray());
 
+enum class ReduceMinMaxMode
+{
+    FIRST_MIN = 0, //!< get index of first min occurrence
+    LAST_MIN  = 1, //!< get index of last min occurrence
+    FIRST_MAX = 2, //!< get index of first max occurrence
+    LAST_MAX  = 3, //!< get index of last max occurrence
+};
+
+/**
+ * @brief Finds indices of min or max elements along provided axis
+ *
+ * @note
+ *      - If input or output array is not continuous, this function will create an internal copy.
+ *      - NaN handling is left unspecified, see patchNaNs().
+ *      - The returned index is always in bounds of input matrix.
+ *
+ * @param src input single-channel array.
+ * @param dst output array of type CV_32SC1 with the same dimensionality as src,
+ * except for axis being reduced - it should be set to 1.
+ * @param mode whether to get the index of first or last occurrence of min or max.
+ * @param axis axis to reduce along.
+ * @sa minMaxLoc, min, max, compare, reduce
+ */
+CV_EXPORTS_W void reduceMinMax(InputArray src, OutputArray dst, ReduceMinMaxMode mode, int axis);
```

Comment:
> reduceMinMax

IMHO, we don't want to mess min/max in that way in public API. Especially with enum.
You don't want to process "both" indices at least.

As suggested on the meeting we should have:
- reduceArgMin
- reduceArgMax

You may add `bool lastIndex = false` optional parameter or add these functions:
- reduceArgMinLast
- reduceArgMaxLast